### PR TITLE
improv: Do not hide `window_group` on Applications screen

### PR DESCRIPTION
**Pull request by ids1024**
This seems to improve the animation opening and closing Applications. It fades cleanly in and out, without the slight visual glitch closing the dock previously.

The only thing is that I'm not sure why Gnome hides `window_group`. It doesn't seem like it should have any visual impact on stock Gnome. Does it eliminate some rendering with a performance impact? Hm...

### CODE REVIEW

**Comment by jacobgkau** [comment]
With this version, I'm seeing the windows disappear for a split second while the apps menu is fading in (possibly less than before any of the animation fixes, but definitely happening). There's also a more obvious point during the fade-out where the windows are hidden while the app menu's opaque background is fading out.

This also causes a regression if I open Applications and then switch directly to Workspaces, where the workspace picker is empty:

Screenshot from 2021-07-06 10-52-54

**Comment by ids1024** [comment]
> This also causes a regression if I open Applications and then switch directly to Workspaces, where the workspace picker is empty:

That should be unrelated and fixed by https://github.com/pop-os/cosmic/pull/117. Rebased the PR to include that.

**Comment by jacobgkau** [comment]
Can confirm the rebase fixed the workspace picker issue, and I'm not sure if I did something wrong before, but now the animation looks perfect too. Going to try and break it, will approve if I can't.

**Comment by jacobgkau** [comment]
In terms of what's left to be improved, the app grid shifts up when switching to Workspaces and the windows shift down when switching to Applications.

https://user-images.githubusercontent.com/7199422/124669255-5395e280-de6f-11eb-84a2-400201453225.mp4

This fade improvement is working as expected, and does not cause any visual glitches on non-primary displays.

**Review by jacobgkau — APPROVED** [review]

**Comment by ids1024** [comment]
> In terms of what's left to be improved, the app grid shifts up when switching to Workspaces and the windows shift down when switching to Applications.

Yeah, I've been looking for a fix for that. It's because of how it Cosmic shows and hides the search bar. (I tried just moving the search bar widget into Applications, but Applications and search results are separate pages...)

**Comment by ids1024** [comment]
One solution to that: instead of hiding the search bar, set its opacity to zero. Then it will still take up space, so there's no shifting. And it could also smoothly fade.

I don't know if we mind having the search bar invisibly allocate vertical space in the Workspaces view.

**Comment by jacobgkau** [comment]
> One solution to that: instead of hiding the search bar, set its opacity to zero. Then it will still take up space, so there's no shifting. And it could also smoothly fade.
> 
> I don't know if we mind having the search bar invisibly allocate vertical space in the Workspaces view.

That seems fine visually, but we definitely don't want type-to-search to work in Workspaces where the search bar isn't visible.

**Comment by DuttaAB-dev** [comment]
> That seems fine visually, but we definitely don't want type-to-search to work in Workspaces where the search bar isn't visible.

I know this is totally unrelated, but why not let that be optional? Or even make typing there make it fall back to the launcher?

**Comment by leviport** [comment]
> I know this is totally unrelated, but why not let that be optional? Or even make typing there make it fall back to the launcher?

This PR isn't the place for such a discussion, but in a nutshell, it is because the functionality would be ambiguous. What would a user be searching for in a workspaces overview, when naming specific workspaces is not supported? If a user is searching for application names, why would that be done in a workspaces overview rather than a launcher?

If this is a discussion you'd like to have, I'd recommend opening a separate issue with a proposal and justifications for that proposal. Such proposals are always welcome.

**Comment by jacobgkau** [comment]
> Or even make typing there make it fall back to the launcher?

I think part of the issue is also that the Launcher and Workspaces can't currently be open at the same time, so typing in Workspaces would close Workspaces, which would be too jarring. If the Launcher was able to open (and highlight windows, etc) within Workspaces, that could be better, but that would require more engineering work. And as both you and Levi have pointed out, that's unrelated to this animation bug-fix PR.

**Comment by ids1024** [comment]
> That seems fine visually, but we definitely don't want type-to-search to work in Workspaces where the search bar isn't visible.

This seems to work as expected by also setting `reactive` to `false` when setting the `opacity` to `0`.

Added a commit with this to the PR.

**Comment by jacobgkau** [comment]
This is the difference in space taken up by the invisible search box in Workspaces:

cosmic-2

The workspace picker being shorter is almost more noticeable than the window area being smaller. This does fix the remaining animation issues:

https://user-images.githubusercontent.com/7199422/124797585-ade68000-df0f-11eb-99a3-d7f05062159e.mp4

It really doesn't make sense to reserve that extra space in Workspaces view given that we're disabling search, but the animation papercuts without it aren't ideal either. It's a shame moving the search box into the Applications page didn't work. Personally, I think this latest version that fixes the animation is the better of the two options, but we might want to get another opinion on that since we've gone through some work to avoid other gaps in Workspaces view.

**Comment by ids1024** [comment]
Yeah, it's a bit unfortunate. If Gnome Shell were our own project, the correct solution would be to modify the widget hierarchy to have the results and applications pages grouped together in their own stack widget, nested inside the stack workspaces is in. Changing that by hotpatching in a shell extension however would be complicated and asking for trouble though.

I wonder if it would be hard to position the search bar widget in such a way that it overlaps with with stack, but then allocating space for it in the applications results pages... a bit of a pain but that might work alright.

**Review by jackpot51 — APPROVED** [review]

**Comment by ids1024** [comment]
Merging. The search bar behavior can be improved in the future if we get something better to work.

**Pull request merged**
